lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -79,43 +78,43 @@ def _handle_key(key):
     return key
 
 
-class TimerTornado(backend_bases.TimerBase):
+class TimerAsyncio(backend_bases.TimerBase):
     def __init__(self, *args, **kwargs):
-        self._timer = None
+        self._task = None
         super().__init__(*args, **kwargs)
 
+    async def _timer_task(self, interval):
+        while True:
+            try:
+                await asyncio.sleep(interval)
+                self._on_timer()
+
+                if self._single:
+                    break
+            except asyncio.CancelledError:
+                break
+
     def _timer_start(self):
         self._timer_stop()
-        if self._single:
-            ioloop = tornado.ioloop.IOLoop.instance()
-            self._timer = ioloop.add_timeout(
-                datetime.timedelta(milliseconds=self.interval),
-                self._on_timer)
-        else:
-            self._timer = tornado.ioloop.PeriodicCallback(
-                self._on_timer,
-                max(self.interval, 1e-6))
-            self._timer.start()
+
+        self._task = asyncio.ensure_future(
+            self._timer_task(max(self.interval / 1_000., 1e-6))
+        )
```

Comment:
Does this run the async loop? Or is it expected to be elsewhere?

Reply:
This will internally use the current event loop with `asyncio.get_event_loop()`

Reply:
But what starts the loop?